src/lib/scheduler/current_scheduler.hpp

```diff
@@ -14,34 +15,38 @@ class AbstractScheduler;
 /**
  * Holds the singleton instance (or the lack of one) of the currently active Scheduler
  */
-class CurrentScheduler {
+class CurrentScheduler : public Noncopyable {
```

Comment:
What is the point of this class? Now that we have `NoScheduler`, all those `if (_instance)` checks should be gone, anyway. Can we get rid of this indirection?